fuel-merkle/src/binary/merkle_tree.rs

```diff
@@ -182,6 +182,11 @@ where
         Ok((root, proof_set))
     }
 
+    pub fn reset(&mut self) {
```

Comment:
does this need to alter storage at all?

Reply:
It actually doesn't because all reads and writes are determined only by the `leaves_count` and `head`. 

However, it does leave the storage in a kind of "dirty" state. It's currently not possible with this interface, but theoretically, with future modifications, we could end up in a position where we request a node that should be empty, and the dirty state returns erroneous data from before the reset. We could introduce another storage trait, e.g. `StorageReset` that we can implement on databases or in-memory storage that allows for implementation specific reset strategies. Then, we can simply call `storage.reset()` here to make sure we have cleaned up entirely.